File: src/main/java/com/time/dev/timetrack/model/Timesheet.java
package com.time.dev.timetrack.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import java.util.Date;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Entity
public class Timesheet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Temporal(TemporalType.DATE)
    private Date date;

    private int hours;

    private String comments;

    @ManyToOne
    @JsonBackReference("user-timesheet")
    @JoinColumn(name = "role_user_id")
    private RoleUser roleUser;

}

File: app/src/app/model/timesheet.ts
export class Timesheet {
  id: number;
  date: Date;
  hours: number;
  comments: string;
  roleUserId: number;

  constructor(timesheet: Partial<Timesheet> = {}) {
    this.id = timesheet.id || 0;
    this.date = timesheet.date ? new Date(timesheet.date) : new Date();
    this.hours = timesheet.hours || 0;
    this.comments = timesheet.comments || '';
    this.roleUserId = timesheet.roleUserId || 0;
  }
}
